extensions/amp-story/1.0/test/test-amp-story.js

```diff
@@ -721,6 +732,30 @@ describes.realWin(
         expect(story.storeService_.get(StateProperty.PAUSED_STATE)).to.be.false;
       });
 
+      it('should show system UI and unpause on resume', async () => {
+        // Fix #28425, on player swipe doesn't get touchend so UI doesn't show.
+        await createStoryWithPages(2, ['cover', 'page-1']);
+
+        story.buildCallback();
+
+        await story.layoutCallback();
+
+        // Touchstarts will hide systemUI and pause story.
+        story.pages_[0].element.dispatchEvent(
+          new TouchEvent('touchstart', getTouchOptions(0, 0))
+        );
+
+        expect(story.storeService_.get(StateProperty.PAUSED_STATE)).to.be.true;
+
+        // Calling resume should resume the paused state and system ui visibility.
+        story.resume_();
+
+        expect(
+          story.storeService_.get(StateProperty.SYSTEM_UI_IS_VISIBLE_STATE)
+        ).to.be.true;
+        expect(story.storeService_.get(StateProperty.PAUSED_STATE)).to.be.false;
+      });
```

Comment:
Simulated error in #28425 
Triggered touchstart (which pauses the story, and sets timer to hide systemUI after 0.5s), then called resume.
Checking for system ui and paused state to return to normal.